### DISCUSSION TITLE
OmniNXT, in position hold the copter runs away, while compass shows right directions

### DISCUSSION THREAD

**Discussion by Chrisi2009** [iNavFlight/inav#8159] [Q&A] [answered]
At first, my build hardware:

Quadcopter with 3 cells LiPo
FC OmniNXT F7
GPS BN-880
SW: INAV4.1

This things working fine...
Flight in angle mode.
The copter is very stable in the air.
After much setups, the compass itself works fine. On the OSD it shwos the right direction in degrees/north/east and so on. Also the OSD shows the staring point and its way to home. On Inav the google map also shows everything wright.

But here are my 2 issues:

1.)
If i switch on position hold, the copter drifts straightly away in random direction.
It never hold its position for a piece of a second.

2.)
If i switch on RTH, the copter lifts up to its setup altitude (in test 5m), turns to north (correctly i think) and then he fly away in 90 degrees CCW (mostly west) and fly away.

What could be wrong ? Why does it fly back in wrong direction, although the OSD shows correct arrow ?

After reading so much in google, my compass is installed correctly, it shows the right degrees and direction to home and so on. (north=0°, east=90°...)

Is there anything i didnt see ?

Chris

**Comment by Jetrell** [comment]
Hi Chris.. Many times the compass can seem correct when level. But in reality it isn't. 
If it was calibrated with localized magnetic interference. Mostly from the quad itself. This can lead to the effect you are experiencing. 
The GPS course can also give the impression that things are okay, with the home arrow pointing in the correct direction.

Try to calibrate it again... Once the calibration is complete, its important that you compare the motion of the quad, to the virtual rendered image on the Configurator Setup page.. Make sure the quad moves the same as that image, on all axis's...When its moving sideways, nose down, nose up etc....  If it doesn't move exactly the same, then its experiencing magnetometer interference.

**Reply by Chrisi2009** [reply]
Hi Jetrell !

Thank you for reply !
So, today i tried to calibrate again and like the last times, everything seems correct.
I also buyed a ordinary mechanical compass...
i thing nobody has it at home :))
With this i compared the directions too. Everything is the same (north/east/south/west).
At the setup page the directions and movements are right too.
But still doesnt work with position hold or rth :(

Could it be, that something in the CLI has to change ? Because so much guys change values and functions in it like i saw at youtube.
But i wouldnt know, what to change.

**Reply by Jetrell** [reply]
Can you paste your `Diff All` setting from the CLI ? 

Just a thought.
Something that is of great importance's... Is to be sure your Flight Control board and your GPS/Magnetometer, both have their ARROW's aligned in the same direction.. Because the IMU (in the FC) and Compass (in the GPS unit) work together.
If you don't, the result can be anything from 'Almost working, to a total failure'.  And this will only effect navigation modes.. Acro and Angle will be fine if its wrong.

But this can be worked around in the settings if you don't want to change the hardware.. It will just take time..

**Comment by Chrisi2009** [comment]
![_20220625_002951](https://user-images.githubusercontent.com/108101402/175715858-af7f3935-c205-447b-9587-04a522dd2a30.JPG)

Here is a picture from my copter,  maybe there is something, i dont see.

**Comment by Chrisi2009** [comment]
Hi Jetrell, here is my CLI setting...

**Comment by Chrisi2009** [comment]
# Diff All

# version
# INAV/OMNIBUSF7NXT [4.1.0](tel:410) Feb [11 2022 / 11](tel:11202211):30:24 [(611](tel:611)d26f4)
# GCC-[10.2.1 20201103](tel:102120201103) (release)

# start the command batch
batch start

# reset configuration to default settings
defaults noreboot

# resources

# mixer

mmix reset

mmix 0  [1.000 -1.000](tel:10001000)  1.000 -1.000
mmix 1  1.000 -1.000 -1.000  1.000
mmix 2  1.000  1.000  1.000  1.000
mmix 3  1.000  1.000 -1.000 -1.000

# servo mix

# servo

# safehome

# logic

# gvar

# pid

# feature
feature -CURRENT_METER
feature -BLACKBOX
feature -AIRMODE
feature GPS
feature PWM_OUTPUT_ENABLE

# beeper

# blackbox
blackbox -NAV_ACC
blackbox NAV_POS
blackbox NAV_PID
blackbox MAG
blackbox ACC
blackbox ATTI
blackbox RC_DATA
blackbox RC_COMMAND
blackbox MOTORS

# map

# serial
serial 5 2 [115200](tel:115200) [115200 0 115200](tel:1152000115200)

# led

# color

# mode_color

# aux
aux [0 0 0 1500 2100](tel:00015002100)
aux [1 1 0 900 2100](tel:1109002100)
aux [2 11 1 1325 1725](tel:211113251725)
aux [3 10 1 1750 2100](tel:310117502100)
aux [4 3 2 900 1300](tel:4329001300)
aux [5 5 2 1325 1725](tel:55213251725)

# adjrange

# rxrange

# temp_sensor

# wp
#wp 0 invalid

# osd_layout
osd_layout 0 0 23 0 H
osd_layout [0 1 22 2](tel:01222) V
osd_layout [0 3 8 6](tel:0386) V
osd_layout [0 7 13 11](tel:071311) V
osd_layout [0 9 2 7](tel:0927) H
osd_layout [0 11 12 7](tel:011127) H
osd_layout [0 12 1 4](tel:01214) H
osd_layout [0 14 2 11](tel:014211) V
osd_layout [0 15 2 2](tel:01522) V
osd_layout [0 22 14 7](tel:022147) V
osd_layout [0 23 12 3](tel:023123) V
osd_layout [0 24 12 9](tel:024129) V
osd_layout [0 28 21 3](tel:028213) H
osd_layout [0 30 1 5](tel:03015) V
osd_layout [0 85 23 11](tel:0852311) V
osd_layout [0 96 2 1](tel:09621) H

# master
set looptime = [500](tel:500)
set gyro_main_lpf_hz = [110](tel:110)
set gyro_main_lpf_type = PT1
set dynamic_gyro_notch_q = [250](tel:250)
set dynamic_gyro_notch_min_hz = [120](tel:120)
set setpoint_kalman_q = [200](tel:200)
set acc_hardware = MPU[6000](tel:6000)
set acczero_x = 69
set acczero_y = -37
set acczero_z = -[219](tel:219)
set accgain_x = [4064](tel:4064)
set accgain_y = [4085](tel:4085)
set accgain_z = [4066](tel:4066)
set align_mag = CW[270](tel:270)FLIP
set mag_hardware = HMC[5883](tel:5883)
set magzero_x = 20
set magzero_y = -[143](tel:143)
set magzero_z = 89
set maggain_x = [513](tel:513)
set maggain_y = [524](tel:524)
set maggain_z = [513](tel:513)
set baro_hardware = LPS25H
set pitot_hardware = ADC
set blackbox_rate_denom = 2
set motor_pwm_protocol = STANDARD
set failsafe_procedure = RTH
set align_board_pitch = 2
set current_meter_type = NONE
set model_preview_type = 3
set applied_defaults = 2
set gps_sbas_mode = EGNOS
set gps_ublox_use_galileo = ON
set airmode_type = THROTTLE_THRESHOLD
set nav_max_auto_speed = [400](tel:400)
set nav_auto_climb_rate = [200](tel:200)
set nav_manual_speed = [400](tel:400)
set nav_manual_climb_rate = [100](tel:100)
set nav_land_slowdown_maxalt = [1000](tel:1000)
set nav_min_rth_distance = [100](tel:100)
set nav_rth_abort_threshold = [100](tel:100)
set nav_rth_altitude = [500](tel:500)
set dji_use_name_for_messages = OFF

# profile
profile 1

set mc_p_pitch = 44
set mc_i_pitch = 75
set mc_d_pitch = 25
set mc_i_roll = 60
set mc_p_yaw = 35
set mc_i_yaw = 80
set dterm_lpf_type = PT3
set d_boost_max =  [1.500](tel:1500)
set antigravity_gain =  [2.000](tel:2000)
set antigravity_accelerator =  [5.000](tel:5000)
set smith_predictor_delay =  [1.500](tel:1500)
set tpa_rate = 20
set tpa_breakpoint = [1200](tel:1200)
set rc_yaw_expo = 70
set roll_rate = 70
set pitch_rate = 70
set yaw_rate = 60

# profile
profile 2


# profile
profile 3


# battery_profile
battery_profile 1

set bat_cells = 3
set battery_capacity = [1800](tel:1800)
set battery_capacity_warning = [450](tel:450)
set battery_capacity_critical = [270](tel:270)
set throttle_idle =  [10.000](tel:10000)

# battery_profile
battery_profile 2


# battery_profile
battery_profile 3


# restore original profile selection
profile 1
battery_profile 1

# save configuration
save

**Reply by Jetrell** [reply]
Nothing really stands out.. for the navigation side.
Concerning what I mentioned yesterday, about the FC board, gyro/acc mounting direction, That has to align with the magnetometer. 
The GPS/mag is mounted correctly, from your picture.. But I can't tell with this twin IMU flight controller.
There was some talk in the past on this board. It may reveal something. https://github.com/iNavFlight/inav/issues/4935 

As a side note.. If your ESC's can run a faster protocol than Standard... That would make it handle better.

**Reply by Chrisi2009** [reply]
I will try Inav version 5 and maybe 3 later, maybe something doesnt work with it.
I will reply after some tests...
Thank you so much Jetrell !!!

**Comment by sensei-hacker** [comment]
Try setting your board yaw to 180 degrees.
As I recall, on that board inav and Betaflight disagree about which edge is the "front".

**Reply by Chrisi2009** [reply]
Oh, thank you Morris, this i will try too today 😊
That coule be too wrong.

**Comment by Chrisi2009** [comment, marked as answer]
So, i think its done...
Finally INAV version 5 works without any issue.
I just installed it as same as version 4 and it works. Position hold and return to home... perfekt !
I didnt change any values on the compass.
Maybe Morris was right with angle change to 180°, but i dont want to look for it right now, version 5 know something better than me :))
I'm just lucky !
Thank you guys for your help and troubleshooting !!!
